framework/data/Sort.php

```diff
@@ -216,7 +230,11 @@ public function getOrders($recalculate = false)
             $definition = $this->attributes[$attribute];
             $columns = $definition[$direction === SORT_ASC ? 'asc' : 'desc'];
             foreach ($columns as $name => $dir) {
-                $orders[$name] = $dir;
+                if (is_int($name)) {
```

Comment:
this deserves a description so it's objective can be easily understood in the future